src/lib/token.test.ts

```diff
@@ -0,0 +1,313 @@
+import {
+  Field,
+  shutdown,
+  isReady,
+  State,
+  state,
+  UInt64,
+  SmartContract,
+  Mina,
+  PrivateKey,
+  Party,
+  method,
+  PublicKey,
+  DeployArgs,
+  Permissions,
+  Token,
+  Ledger,
+} from '../../dist/server';
+
+const tokenSymbol = 'MY_TOKEN';
+
+class TokenContract extends SmartContract {
+  @state(UInt64) totalAmountInCirculation = State<UInt64>();
+  @state(UInt64) maxAmountInCirculation = State<UInt64>();
+
+  deploy(args: DeployArgs) {
+    super.deploy(args);
+    this.setPermissions({
+      ...Permissions.default(),
+      editState: Permissions.proofOrSignature(),
+    });
+  }
+
+  @method init() {
+    this.tokenSymbol.set(tokenSymbol);
+    this.totalAmountInCirculation.set(UInt64.zero);
```

Comment:
I think we should consider using the term Quantity, instead of Amount, everywhere. On trading dashboards, qty refers to the number of tokens and amount prefers to `price * amount`, which would be USD etc. when trading on fiat markets. So qty is more consistent with how it's used on most financial platforms.

Reply:
I also typically see (and we use consistently in the backend code) the term "totalSupply" for this -- probably for consistency we should go with that.